diff --git a/src/adapters/env.ts b/src/adapters/env.ts
index 034b982..bfa5d93 100644
--- a/src/adapters/env.ts
+++ b/src/adapters/env.ts
@@ -31,6 +31,7 @@ interface Env {
   VITE_REPORT_FORM_PLATFORM_ENTRY?: string;
   VITE_REPORT_FORM_URL?: string;
   VITE_REPORT_FORM_URL_ENTRY?: string;
+  VITE_RESOLVE_RELATIVE_URLS?: string;
 }

 type GetFiatExchangeRatesEnvParams = Pick<
@@ -180,6 +181,7 @@ const envToDomain = ({
   VITE_REPORT_FORM_PLATFORM_ENTRY,
   VITE_REPORT_FORM_URL,
   VITE_REPORT_FORM_URL_ENTRY,
+  VITE_RESOLVE_RELATIVE_URLS,
 }: Env): Promise<domain.Env> => {
   const polygonZkEVMNetworkId = z.coerce.number().positive().parse(VITE_POLYGON_ZK_EVM_NETWORK_ID);
   const isOutdatedNetworkModalEnabled = stringBooleanParser.parse(
@@ -188,7 +190,30 @@ const envToDomain = ({
   const forceUpdateGlobalExitRootForL1 = stringBooleanParser.parse(
     VITE_ETHEREUM_FORCE_UPDATE_GLOBAL_EXIT_ROOT
   );
-  const bridgeApiUrl = VITE_BRIDGE_API_URL;
+
+  const isSupportForRelativeUrlsEnabled = stringBooleanParser.parse(VITE_RESOLVE_RELATIVE_URLS);
+
+  let bridgeApiUrl: string;
+  if (isSupportForRelativeUrlsEnabled) {
+    bridgeApiUrl = window.location.origin + VITE_BRIDGE_API_URL;
+  } else {
+    bridgeApiUrl = VITE_BRIDGE_API_URL;
+  }
+
+  let ethereumRpcUrl: string;
+  if (isSupportForRelativeUrlsEnabled) {
+    ethereumRpcUrl = window.location.origin + VITE_ETHEREUM_RPC_URL;
+  } else {
+    ethereumRpcUrl = VITE_ETHEREUM_RPC_URL;
+  }
+
+  let zkevmRpcUrl: string;
+  if (isSupportForRelativeUrlsEnabled) {
+    zkevmRpcUrl = window.location.origin + VITE_POLYGON_ZK_EVM_RPC_URL;
+  } else {
+    zkevmRpcUrl = VITE_POLYGON_ZK_EVM_RPC_URL;
+  }
+
   const outdatedNetworkModal: domain.Env["outdatedNetworkModal"] = isOutdatedNetworkModalEnabled
     ? {
         isEnabled: true,
@@ -208,13 +233,13 @@ const envToDomain = ({
       explorerUrl: VITE_ETHEREUM_EXPLORER_URL,
       poeContractAddress: VITE_ETHEREUM_PROOF_OF_EFFICIENCY_CONTRACT_ADDRESS,
       rollupManagerAddress: VITE_ETHEREUM_ROLLUP_MANAGER_ADDRESS,
-      rpcUrl: VITE_ETHEREUM_RPC_URL,
+      rpcUrl: ethereumRpcUrl,
     },
     polygonZkEVM: {
       bridgeContractAddress: VITE_POLYGON_ZK_EVM_BRIDGE_CONTRACT_ADDRESS,
       explorerUrl: VITE_POLYGON_ZK_EVM_EXPLORER_URL,
       networkId: polygonZkEVMNetworkId,
-      rpcUrl: VITE_POLYGON_ZK_EVM_RPC_URL,
+      rpcUrl: zkevmRpcUrl,
     },
   }).then((chains) => {
     const ethereumChain = chains.find((chain) => chain.key === "ethereum");
@@ -250,7 +275,7 @@ const envToDomain = ({
 const envParser = StrictSchema<Env, domain.Env>()(
   z
     .object({
-      VITE_BRIDGE_API_URL: z.string().url(),
+      VITE_BRIDGE_API_URL: z.string(),
       VITE_ENABLE_DEPOSIT_WARNING: z.string(),
       VITE_ENABLE_FIAT_EXCHANGE_RATES: z.string(),
       VITE_ENABLE_OUTDATED_NETWORK_MODAL: z.string().optional(),
@@ -260,7 +285,7 @@ const envParser = StrictSchema<Env, domain.Env>()(
       VITE_ETHEREUM_FORCE_UPDATE_GLOBAL_EXIT_ROOT: z.string(),
       VITE_ETHEREUM_PROOF_OF_EFFICIENCY_CONTRACT_ADDRESS: z.string().length(42),
       VITE_ETHEREUM_ROLLUP_MANAGER_ADDRESS: z.string().length(42),
-      VITE_ETHEREUM_RPC_URL: z.string().url(),
+      VITE_ETHEREUM_RPC_URL: z.string(),
       VITE_FIAT_EXCHANGE_RATES_API_KEY: z.string().optional(),
       VITE_FIAT_EXCHANGE_RATES_API_URL: z.string().url().optional(),
       VITE_FIAT_EXCHANGE_RATES_ETHEREUM_USDC_ADDRESS: z.string().length(42).optional(),
@@ -271,11 +296,12 @@ const envParser = StrictSchema<Env, domain.Env>()(
       VITE_POLYGON_ZK_EVM_BRIDGE_CONTRACT_ADDRESS: z.string().length(42),
       VITE_POLYGON_ZK_EVM_EXPLORER_URL: z.string().url(),
       VITE_POLYGON_ZK_EVM_NETWORK_ID: z.string(),
-      VITE_POLYGON_ZK_EVM_RPC_URL: z.string().url(),
+      VITE_POLYGON_ZK_EVM_RPC_URL: z.string(),
       VITE_REPORT_FORM_ERROR_ENTRY: z.string().optional(),
       VITE_REPORT_FORM_PLATFORM_ENTRY: z.string().optional(),
       VITE_REPORT_FORM_URL: z.string().optional(),
       VITE_REPORT_FORM_URL_ENTRY: z.string().optional(),
+      VITE_RESOLVE_RELATIVE_URLS: z.string().optional(),
     })
     .transform(envToDomain)
 );
